demos/TicTacToe/Classes/main_menu_scene.cc

```diff
@@ -360,64 +376,73 @@ bool MainMenuScene::init() {
-  join_text_field->setPosition(420, 45);
-  join_text_field->setAnchorPoint(Vec2(0, 0));
-  join_text_field->setColorSpaceHolder(Color3B::WHITE);
-  join_text_field->setDelegate(this);
-
-  auto join_text_field_border = Sprite::create(kTextFieldBorderImage);
-  join_text_field_border->setPosition(390, 50);
-  join_text_field_border->setAnchorPoint(Vec2(0, 0));
-  join_text_field_border->setScale(.53f);
-  this->addChild(join_text_field_border, /*layer_index=*/0);
-
-  // Create a touch listener to handle the touch event.
-  auto join_text_field_touch_listener = EventListenerTouchOneByOne::create();
-
-  join_text_field_touch_listener->onTouchBegan =
-      [join_text_field, this](cocos2d::Touch* touch,
-                              cocos2d::Event* event) -> bool {
-    // Returns false, not consuming the event, to exit the layer if
-    // current_state_ is not in the kGameMenuState or is switching states.
-    if (previous_state_ != current_state_ || current_state_ != kGameMenuState) {
-      return false;
-    }
-    const auto bounds = event->getCurrentTarget()->getBoundingBox();
-    const auto point = touch->getLocation();
-    if (bounds.containsPoint(point)) {
-      // Show the on screen keyboard and places character inputs into the text
-      // field.
-      auto str = join_text_field->getString();
-      auto text_field = dynamic_cast<TextFieldTTF*>(event->getCurrentTarget());
-      text_field->setCursorEnabled(true);
-      text_field->attachWithIME();
-    } else {
-      auto text_field = dynamic_cast<TextFieldTTF*>(event->getCurrentTarget());
-      text_field->setCursorEnabled(false);
-      text_field->detachWithIME();
-    }
 
-    return true;
-  };
+  auto join_text_field_position = Size(480, 95);
+  auto join_text_field_size = Size(180, 80);
+  auto join_text_field = TextField::create("code", "Arial", 48);
+  join_text_field->setTextHorizontalAlignment(TextHAlignment::CENTER);
+  join_text_field->setPosition(join_text_field_position);
+  join_text_field->setAnchorPoint(Vec2::ANCHOR_MIDDLE);
+  join_text_field->setTouchSize(join_text_field_size);
+  join_text_field->setTouchAreaEnabled(true);
+  join_text_field->setMaxLength(/*max_characters=*/4);
+  join_text_field->setMaxLengthEnabled(true);
+
+  // Adds the event listener to handle the actions for a textfield.
+  join_text_field->addEventListener(
+      [this](Ref* sender, TextField::EventType type) {
+        auto join_text_field = dynamic_cast<TextField*>(sender);
+        string join_text_field_string = join_text_field->getString();
+        // Transforms the letter casing to uppercase.
+        std::transform(
+            join_text_field_string.begin(), join_text_field_string.end(),
+            join_text_field_string.begin(),
+            [](unsigned char c) -> unsigned char { return std::toupper(c); });
+        // Creates a repeating blink action for the cursor.
+        const auto repeated_blink = CreateBlinkingCursorAction(join_text_field);
```

Comment:
since this is only ever used in the Attach_with_ime case block, better to just do this in the case block:

```
join_text_field->runAction(CreateBlinkingCursorAction(join_text_field));
```